src/corebluetooth/peripheral.rs

```diff
@@ -55,39 +55,76 @@ impl Peripheral {
     ) -> Self {
         // Since we're building the object, we have an active advertisement.
         // Build properties now.
-        let properties = PeripheralProperties {
+        let properties = Arc::new(Mutex::from(PeripheralProperties {
             // Rumble required ONLY a BDAddr, not something you can get from
             // MacOS, so we make it up for now. This sucks.
             address: uuid_to_bdaddr(&uuid.to_string()),
             address_type: AddressType::Random,
             local_name: local_name,
             tx_power_level: None,
             manufacturer_data: HashMap::new(),
+            service_data: HashMap::new(),
+            services: Vec::new(),
             discovery_count: 1,
             has_scan_response: true,
-        };
+        }));
         let notification_handlers = Arc::new(Mutex::new(Vec::<NotificationHandler>::new()));
         let nh_clone = notification_handlers.clone();
+        let p_clone = properties.clone();
+        let m_clone = manager.clone();
         task::spawn(async move {
             let mut event_receiver = event_receiver;
             loop {
-                let event = event_receiver.next().await;
-                if event.is_none() {
+                if let Some(event) = event_receiver.next().await {
```

Comment:
This match statement was changed from an `Option<T>` to just `T`. so that each variant can be caught at compile time and not only at runtime only when someone is paying attention to it.